mage_ai/orchestration/pipeline_scheduler.py

```diff
@@ -472,32 +486,156 @@ def __schedule_blocks(self, block_runs: List[BlockRun] = None) -> None:
                 self.__build_tags(**tags),
             )
 
-    def __schedule_integration_pipeline(self, block_runs: List[BlockRun] = None) -> None:
-        if job_manager.has_pipeline_run_job(self.pipeline_run.id):
-            return
-
-        block_runs_to_schedule = self.initial_block_runs if block_runs is None else block_runs
-        block_runs_to_schedule = self.__fetch_crashed_block_runs() + block_runs_to_schedule
+    def __schedule_integration_streams(self, block_runs: List[BlockRun] = None) -> None:
+        if block_runs is not None:
+            block_runs_to_schedule = block_runs
+        else:
+            # Fetch all "in progress" blocks to handle crashed block runs
```

Comment:
This code below doesn't fetch crashed block runs. It just fetches blocks that are not completed.

Reply:
yes, this is on purpose to handle cases where block runs never completed because their parent process crashed.